lib/react.dart

```diff
@@ -1766,6 +1780,11 @@ ComponentRegistrar2 registerComponent2 = (
   throw new Exception('setClientConfiguration must be called before registerComponent.');
 };
 
+/// Registers [componentFactory] on both client and server.
```

Comment:
This doc comment looks like copypasta from `ComponentRegistrar`. At the very least, the bit about it registering stuff for server side rendering should be removed.  It would be even better if the comment fully explained how it should be used.